server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -967,17 +968,9 @@ private CompletionStage<RestResponse> getCacheConfig(RestRequest request) {
       if (!invocationHelper.getRestCacheManager().getInstance().getCacheConfigurationNames().contains(cacheName)) {
          responseBuilder.status(NOT_FOUND).build();
       }
-      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, request);
-      if (cache == null)
-         return invocationHelper.newResponse(request, NOT_FOUND).toFuture();
-
-      AuthorizationManager authorizationManager = SecurityActions.getCacheAuthorizationManager(cache.getAdvancedCache());
-      if (authorizationManager == null || authorizationManager.isPermissive()) {
-         // Cache is not secured, use the global authz
-         invocationHelper.getRestCacheManager().getAuthorizer().checkPermission(AuthorizationPermission.ADMIN);
-      }
-      Configuration cacheConfiguration = cache.getCacheConfiguration();
-
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
+      EmbeddedCacheManager subjectCacheManager = cacheManager.withSubject(request.getSubject());
+      Configuration cacheConfiguration = SecurityActions.getCacheConfiguration(subjectCacheManager, cacheName);
```

Comment:
Do we want provide cache configurations also to non admin?

Reply:
@fax4ever no, and we aren't providing it. the endpoint is secured for admins

Reply:
you're right... I didn't get it